File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int idx = 0;
        for (int i = 0; i < score.length; i++) {
            if (i < k) {
                pq.add(score[i]);
            } else {
                if (pq.peek() < score[i]) {
                    pq.poll();
                    pq.add(score[i]);
                }
            }
            answer[idx++] = pq.peek();
        }
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
from heapq import heappush, heappop

def solution(k, score):
    heap, answer = [], []
    for i in range(len(score)):
        if i < k:
            heappush(heap, score[i])
        else:
            if heap[0] < score[i]:
                heappop(heap)
                heappush(heap, score[i])
        answer.append(heap[0])
    return answer